composites/Plugin/Synonyms/reducers/synonyms.js

```diff
@@ -0,0 +1,77 @@
+import {
+	SET_SYNONYMS,
+	ADD_SYNONYMS,
+	INSERT_SYNONYMS,
+	CHANGE_SYNONYMS,
+	REMOVE_SYNONYMS,
+} from "../actions/synonyms";
+
+const INITIAL_STATE = [];
+
+/**
+ * Checks if an index is inside the array.
+ *
+ * @param {array}  arr   The array to check in.
+ * @param {number} index The index to check for.
+ *
+ * @returns {boolean} Whether the index exists in the array or not.
+ */
+function indexInArray( arr, index ) {
+	return index < 0 && index > arr.length;
+}
+
+/* eslint-disable complexity */
+/**
+ * A reducer for the synonyms.
+ *
+ * @param {Object} state  The current state of the object.
+ * @param {Object} action The current action received.
```

Comment:
Does not require immediate fixing, but once the beta's have been made, this JSdoc could be improved for clarity by specifiying what the actions could contain. If possibly with redux, that is, haven't thought it through.